src/plugins/intel_gpu/src/runtime/ocl/ocl_memory.cpp

```diff
@@ -439,6 +443,10 @@ event::ptr gpu_usm::fill(stream& stream) {
 }
 
 event::ptr gpu_usm::copy_from(stream& stream, const memory& other, bool blocking) {
+    if (_bytes_count == 0) {
+        GPU_DEBUG_TRACE_DETAIL << "Skip EnqueueMemcpy for 0 size tensor" << std::endl;
+        return nullptr;
```

Comment:
I think it's safer to return user event instead of nullptr, at least while we have `event->wait()` calls in plugin (w/o any pointer validity checks)

Reply:
I wanted to reduce the even creation overhead too, but I agree safeness is important too. I will update.

Reply:
Yeah, agree, not sure how much it affects performance, but if the impact is indeed noticeable, we may consider to implement all wait calls with stream::wait_for_events() and check pointers there